### DISCUSSION TITLE
Using on-prem PGVector as Vector store for Oracle / PL/SQL based embeddings & training?

### DISCUSSION THREAD

**Discussion by tomthebuzz** [vanna-ai/vanna#384] [Q&A]
I am in a situation where I would need to use an on-premise / air-gapped PGVector instance as Vector Store for the training  / introspection / embeddings of quite a large Oracle PL/SQL based legacy DWH. Has anyone gotten this setup to work? 

As Vanna hosted supports PGVec out of the box it should be quite feasible in principle.

Any suggestion or directional idea is welcome as we are currently still in an experimental stage.

**Comment by npradeep357** [comment]
Right now the default implementation what Vanna provides will not work with on-premise pgvector instance. 
You need to write your own.

**Comment by Deepfarkade** [comment]
> I am in a situation where I would need to use an on-premise / air-gapped PGVector instance as Vector Store for the training / introspection / embeddings of quite a large Oracle PL/SQL based legacy DWH. Has anyone gotten this setup to work?
> 
> As Vanna hosted supports PGVec out of the box it should be quite feasible in principle.
> 
> Any suggestion or directional idea is welcome as we are currently still in an experimental stage.

Please follow the below link where the pull request is there for pgvector - 
https://github.com/vanna-ai/vanna/pull/469/commits/f7e66a5798d86b88a7e9d86b574d826772cf05b8#diff-41f15d32fcaad4726f17ac2725de38350775fb745a55e7600f9cebf213f38c5b
